src/Region/RegionHandler.cc

```diff
@@ -2285,165 +2285,87 @@ casacore::Vector<float> RegionHandler::GetTemporaryRegionProfile(
 void RegionHandler::GetStokesPtotal(
     const ProfilesMap& profiles_q, const ProfilesMap& profiles_u, const ProfilesMap& profiles_v, ProfilesMap& profiles_ptotal) {
     auto calc_step1 = [&](double q, double u) {
-        if (!std::isnan(q) && !isnan(u)) {
-            return (pow(q, 2) + pow(u, 2));
-        }
-        return std::numeric_limits<double>::quiet_NaN();
+        return (IsValid(q, u) ? (std::pow(q, 2) + std::pow(u, 2)) : std::numeric_limits<double>::quiet_NaN());
```

Comment:
I was suggesting eliminating the repeated check entirely. I think that you could do that by creating a function that takes a calculation function as a parameter and returns a function with the same signature which does the check and then either returns the result of the original function or NaN. But that may be too convoluted, and this already looks much more legible, so I'm happy to leave it as it is.

Reply:
Actually, I think that you can do this quite cleanly by defining a new lambda inside your `CombineStokes` function. Instead of using `func` directly, you could define a `func_if_valid` which does the check and either calls `func` on the operands or returns NaN. Then you could use `func_if_valid` in the transform, and you could strip out the checks from all the original lambdas so that they contain only the unique calculation code.